src/System.Net.Http/src/System/Net/Http/HttpClient.cs

```diff
@@ -614,7 +614,7 @@ private static void CheckBaseAddress(Uri baseAddress, string parameterName)
             TaskCompletionSource<HttpResponseMessage> tcs)
         {
             LogSendError(request, cancellationTokenSource, "SendAsync", null);
-            tcs.TrySetCanceled();
+            tcs.TrySetCanceled(cancellationTokenSource.Token);
```

Comment:
@CIPop I have a bug assigned to me that is basically this issue but on Desktop.  We should always have had this behavior (passing in the right Token) but it's been wrong for a while.  Currently on Desktop (and CoreFx without this fix), the wrong Token is put into the OperationCanceledException.  So, passing in this Token properly will ;propagate it into the OperationCanceledException.